common/src/lib.rs

```diff
@@ -84,6 +84,9 @@ pub type Gas = u64;
 pub trait Origin: Sized {
     fn into_origin(self) -> H256;
     fn from_origin(val: H256) -> Self;
+    fn cast<T: Origin>(self) -> T {
```

Comment:
Isn't it just `into()`?

Reply:
https://github.com/gear-tech/gear/pull/3567#discussion_r1422611930